third_party/ibis/ibis_impala/api.py

```diff
@@ -193,12 +194,16 @@ def _if_null(op):
     return ops.Coalesce((op.arg, op.ifnull_expr))
 
 
+@lru_cache(maxsize=2)
 def _get_schema_using_query(self, query):
     # Removing LIMIT 0 around query since it returns no results in Hive
-    cur = self.raw_sql(f"SELECT * FROM ({query}) t0 LIMIT 1")
-    cur.fetchall()
+    cur = self.raw_sql(query + " LIMIT 1")
```

Comment:
Will this cause an issue if the custom query provided already has a LIMIT function within it? We might need to add a check - if it does already have a LIMIT, we can replace it with 'LIMIT 1'.

Reply:
It appears to me that the existing code `f"SELECT * FROM ({query}) t0 LIMIT 1"` does the same thing as the proposed change and will work correctly if the query has a limit clause. The proposed change will likely throw a syntax error if the custom query had a limit clause.

Reply:
@sundar-mudupalli-work, yes the earlier proposed solution will throw a syntax error if custom-query has a LIMIT keyword, so the `update_query_with_limit` does a regex check that if the query has a LIMIT function, and it changes it to LIMIT 1, if it has else we add LIMIT 1 to the query, to fetch the schema.